packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -534,4 +535,21 @@ export class ProjectDataSource {
 
     return _.isEqual(specPattern, [component])
   }
+
+  async maybeGetProjectId (source: ProjectShape) {
+    // If this is the currently active project, we can look at the project id
+    if (source.projectRoot === this.ctx.currentProject) {
+      return await this.projectId()
+    }
+
+    // Get the saved state & resolve the lastProjectId
+    const savedState = await source.savedState?.()
+
+    if (savedState?.lastProjectId) {
+      return savedState.lastProjectId
+    }
+
+    // Otherwise, we can try to derive the projectId by reading it from the config file (todo)
```

Comment:
Do we need to do this todo?

Reply:
Thoughts, @tgriesser?

Reply:
This TODO doesn't make sense, we already tried this on one 544 - if it's not there, it's not available, unless they are in global mode.

Why do we even need to do this here? Eg - why grab from the savedState at all? 

The only use case I can think of is pre-fetching runs for the previous project, when we are launching in global mode and before we've chosen a project, but this seems like an optimization that's unlikely to be super useful. Why not just remotely fetch when they choose a project?